### DISCUSSION TITLE
IGMA (In-Grain Misorientation Axis)

### DISCUSSION THREAD

**Discussion by JSLecomte** [mtex-toolbox/mtex#1880] [Ask Anything]
There is already a discussion on **IGMA** (_in-Grain Misorientation Axis_): but the answer does not satisfy me.
Through an example (here titanium), you will be able to redo my calculations and thus better see where my error lies.
I'm going to use the titanium map from mtex. I charge it, visualize it.
```matlab
mtexdata titanium
% segment grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',5*degree);

% remove two pixel grains
ebsd(grains(grains.grainSize<=2)) = [];
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',5*degree);
ebsd=ebsd('Titanium (Alpha)');
% smooth them
grains = grains.smooth(5);
figure
ipfKey = ipfColorKey(ebsd);
colors = ipfKey.orientation2color(ebsd.orientations);
plot(ebsd,colors)
text(grains,grains.id,'FontSize', 10)

```
![figure1](https://github.com/mtex-toolbox/mtex/assets/89849334/e852c1b1-290f-49b1-9d91-36749e92d2a9)

For my example, I'm going to look the grain 57.
We can visualize this grain, the intragranular disorientation (`mis2mean`) and also the KAM
```matlab
figure
plot(ebsd(grains(57)),ebsd(grains(57)).orientations)
hold on
plot(grains(57).boundary,'lineWidth',2)
hold off

mtexTitle('Grain 57');
figure
plot(ebsd(grains(57)),ebsd(grains(57)).mis2mean.angle./degree)
mtexColorMap hot
mtexColorbar('location','southOutSide','title','mis. Angle [°]')
hold on
plot(grains(57).boundary,'edgecolor','k','linewidth',.5)
hold off
mtexTitle('mis2mean');

% KAM
figure
KAM = ebsd(grains(57)).KAM;
plot(ebsd(grains(57)),KAM);
mtexColorbar('location','southOutSide','title','KAM [°]')
hold on
plot(grains(57).boundary,'linewidth',2)
hold off
mtexTitle('KAM');
```
![figure5](https://github.com/mtex-toolbox/mtex/assets/89849334/cca6eed7-65a9-4df2-98a6-b4b2249a5a85)

---
Now we can visualize the misorientation angle histogram and also the misorientation axis for grain 57)
```
figure
mori=ebsd(grains(57)).mis2mean;
plotAngleDistribution(mori,'DisplayName','grain 57')
%%
figure
plotAxisDistribution(mori,'smooth')
mdf = calcMDF(mori);
[value,ori] = max(mdf,'numLocal',3)
annotate(ori.axis)
```
![figure6](https://github.com/mtex-toolbox/mtex/assets/89849334/5d8f59fd-f7e8-4dfc-8b73-f25ac6fdbf9c)

**HERE IS MY PROBLEM:** 
I am looking for the maximum intensity of my inverse pole figure (first, I calculate a disorientation function with `calcMDF`)
and as you can see, I ask him to find the 3 maxima and they are not placed in the right place: **where is my mistake?**
thanks in advance

**Comment by zstar-kkk** [comment]
figure
mori=ebsd(grains(57)).mis2mean;
plotAngleDistribution(mori,'DisplayName','grain 57')
%%
figure
plotAxisDistribution(mori,'smooth')

moriaxis = mori.axis;
moriaxis.opt.plot = 1;
figure;
h = plot(moriaxis,'smooth','antipodal','fundamentalRegion');

cx = h.XData;
cy = h.YData;
cz = h.ZData;
maxvalue = max(max(cz));
[row col] = find(cz==maxvalue);
maxcx = cx(row,col);
maxcy = cy(row,col);

dt = datatip(h,maxcx,maxcy);
tipc = dt.Content
tipc = tipc{1}

tipc_arr = strsplit(tipc, {' ', '[', ']'});
num1 = str2double(tipc_arr{end-3});
num2 = str2double(tipc_arr{end-2});
num3 = str2double(tipc_arr{end-1});

maxaxis = Miller(num1,num2,num3,ebsd.CS)
